sources/assets/Stride.Core.Assets.Quantum/AssetNodeContainer.cs

```diff
@@ -33,10 +34,35 @@ public AssetNodeContainer()
         NodeBuilder.RegisterPrimitiveType(typeof(Matrix));
         NodeBuilder.RegisterPrimitiveType(typeof(UPath));
         NodeBuilder.RegisterPrimitiveType(typeof(AngleSingle));
+
+        var registeredTypes = new HashSet<Type>();
         // Register content types as primitive so they are not processed by Quantum
         foreach (var contentType in AssetRegistry.GetContentTypes())
         {
             NodeBuilder.RegisterPrimitiveType(contentType);
+            registeredTypes.Add(contentType);
         }
+
+        AssemblyRegistry.AssemblyRegistered += (sender, args) =>
```

Comment:
We might need to make `AssetNodeContainer` be `IDisposable` and unregister to those events when disposing. I'm thinking this might otherwise cause some issues when re-opening a projhect or opening a new project from within the Game Studio (without closing it).

Reply:
I ended up fixing this one in a different way as reaching it to call disposable ended up creating a significant amount of garbage logic
let me know what you think

Reply:
Not sure how, but would be nice to benchmark (or at least estimate) the performance of this version compared to before.

Reply:
The only way it could be slower is if the type being tested has close to or more base types than the amount of content types registered;
`AssetRegistry.IsExactContentType` does one hashtable test per base types the type has, while the `RegisterPrimitiveType` version does one `Type.IsAssignableFrom` tests for each registered content types.

I kept the rest of the logic as is, but it definitely can be optimized further if you think this would be necessary ?